packages/table/src/V11Adapter/V11Adapter.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { ReactElement, useMemo, useRef, useState } from 'react';
+import { VirtualItem } from 'react-virtual';
+
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import { Cell, HeaderCell } from '../Cell';
+import ExpandedContent from '../ExpandedContent/ExpandedContent';
+import HeaderRow from '../HeaderRow';
+import Row from '../Row';
+import TableBody from '../TableBody';
+import TableContainer from '../TableContainer';
+import TableHead from '../TableHead';
+import useLeafyGreenTable, {
+  LeafyGreenTableCell,
+  LeafyGreenTableRow,
+  LGColumnDef,
+  LGRowData,
+  LGTableDataType,
+} from '../useLeafyGreenTable';
+import Table, { flexRender, getCoreRowModel, getSortedRowModel } from '..';
+
+import processColumns from './processColumns';
+import processData from './processData';
+import { V11AdapterProps } from './V11Adapter.types';
+
+// assumes table is first element in children
+// reads columns from columns' keys
+// supports up to one layer of nested rows
+// assumes that the key of a column in the original data === column label header in lowercase; can be overridden by `headerLabelMapping`
+const V11Adapter = <T extends LGRowData>({
+  children,
+  shouldAlternateRowColor,
+  useVirtualScrolling = false,
+  hasSelectableRows = false,
+  headerLabelMapping,
+}: V11AdapterProps<T>) => {
+  const { darkMode } = useDarkMode();
+  const containerRef = useRef(null);
+  const OldTable = React.Children.toArray(children)[0];
+  const {
+    data,
+    columns,
+    children: childrenFn,
+  } = (OldTable as ReactElement).props;
+  const processedColumns: Array<LGColumnDef<T>> = useMemo(
+    () => processColumns(data, columns, headerLabelMapping),
+    [data, columns, headerLabelMapping],
+  );
+  const processedData: Array<LGTableDataType<T>> = useState(() =>
```

Comment:
doesn't useState return a tuple?

Reply:
same

Reply:
Yeah I did this because then the type would be: `const [processedData,]: [Array<LGTableDataType<T>>, Dispatch<...>] ` instead of what it is now. It's really a tradeoff of whether we want the `[0]` to be more clear or the type to be easier to decipher. I leaned more to the type clarity personally, but I'll change it and see if y'all disagree

Reply:
You can pass a type into `useState` itself to get the types
```
const processedData = useState<Array<LGTableDataType<T>>>(...)
```